NewsApp/NewsApp/Controller/NewsController.swift

```diff
@@ -9,38 +9,43 @@
 import UIKit
 
 class NewsController: UITableViewController {
-    private let apiClient = APIClient()
-    private var newsArticles: [Article] = []
+    private let newsService: NewsService = NewsServiceImpl()
+    var category: Category?
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        apiClient.send(GetTopHeadlines()) { [weak self] result in
-            switch result {
-            case .success(let response):
-                let articles = response.articles ?? []
-                self?.newsArticles = articles
-                DispatchQueue.main.async {
-                    self?.tableView.reloadData()
-                }
-            case .failure(let error):
-                debugPrint(error.localizedDescription)
+        self.refreshControl?.addTarget(self, action: #selector(reloadDataTableView), for: UIControl.Event.valueChanged)
```

Comment:
Pour plus tard mais .valueChanged suffit normalement ;-)

Reply:
fix in "fix NewsController"